src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -29,25 +39,95 @@ export default class FileField extends React.Component {
     }
   }
 
-  onAddFile = (event, index = null) => {
+  componentDidMount() {
+    // The File object is not preserved in the save-in-progress data
+    // We need to remove these entries; an empty `file` is included in the
+    // entry, but if API File Object still exists (within the same session), we
+    // can't use Object.keys() on it because it returns an empty array
+    const formData = (this.props.formData || []).filter(
+      // keep - file may not exist (already uploaded)
+      // keep - file may contain File object; ensure name isn't empty
+      // remove - file may be an empty object
+      data => !data.file || (data.file?.name || '') !== '',
+    );
+    if (formData.length !== (this.props.formData || []).length) {
+      this.props.onChange(formData);
+    }
+  }
```

Comment:
No, it's related to the password code.

What happens is when a user chooses an encrypted PDF, it adds an entry in `formData` that includes a `file` value (this is the [browser API `File` object](https://developer.mozilla.org/en-US/docs/Web/API/File/File)). But if a user decides to finish the form later, the `File` object becomes an empty object within the save-in-progress data. Upon returning to the page, the empty file object would prevent them from using the same entry to add a password because there is no `File` object pointing to their local device. This code removes the empty `file` object.